pkg/provider/aws/resources/kms.go

```diff
@@ -0,0 +1,172 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/sanitization/aws"
+)
+
+const (
+	KMS_KEY_TYPE         = "kms_key"
+	KMS_ALIAS_TYPE       = "kms_alias"
+	KMS_REPLICA_KEY_TYPE = "kms_replica_key"
+)
+
+type (
+	KmsKey struct {
+		Name                string
+		ConstructsRef       core.AnnotationKeySet
+		Description         string
+		Enabled             bool
+		EnableKeyRotation   bool
+		KeyPolicy           *PolicyDocument
+		KeySpec             string
+		KeyUsage            string
+		MultiRegion         bool
+		PendingWindowInDays int
+	}
+
+	KmsAlias struct {
+		Name          string
+		ConstructsRef core.AnnotationKeySet
+		AliasName     string
+		TargetKey     *KmsKey
+	}
+
+	KmsReplicaKey struct {
+		Name                string
+		ConstructsRef       core.AnnotationKeySet
+		Description         string
+		Enabled             bool
+		KeyPolicy           *PolicyDocument
+		PendingWindowInDays int
+		PrimaryKey          *KmsKey
+	}
+)
+
+type KmsKeyCreateParams struct {
+	AppName string
+	Refs    core.AnnotationKeySet
+	Name    string
+}
+
+func (key *KmsKey) Create(dag *core.ResourceGraph, params KmsKeyCreateParams) error {
+
+	name := aws.KmsKeySanitizer.Apply(fmt.Sprintf("%s-%s", params.AppName, params.Name))
+	key.Name = name
+	key.ConstructsRef = params.Refs
+
+	existingKey, found := core.GetResource[*KmsKey](dag, key.Id())
+	if found {
+		existingKey.ConstructsRef.AddAll(params.Refs)
+		return nil
+	}
+	dag.AddResource(key)
+	return nil
+}
+
+type KmsKeyConfigureParams struct {
+}
+
+func (key *KmsKey) Configure(params KmsKeyConfigureParams) error {
+	key.EnableKeyRotation = true
+	key.Enabled = true
+	key.MultiRegion = false
+	key.KeySpec = "SYMMETRIC_DEFAULT"
+	key.KeyUsage = "ENCRYPT_DECRYPT"
```

Comment:
can we `const`-ify these two?

Reply:
we could, but these fields could be configured in the future to have different values which is why i didnt do that for now